util-m5: Add default M5OP_ADDR to arm64

hnpl opened the pull request:
As pointed out here [1], the expected M5OP_ADDR for arm64 arch is 0x10010000. This change reflects that.

[1] https://github.com/gem5/gem5/pull/725

Change-Id: I7e72f5ea20d4aacf3115a485ba7cd664d33d037e

hnpl committed:
util-m5: Add default M5OP_ADDR to arm64

As pointed out here [1], the expected M5OP_ADDR for arm64 arch
is 0x10010000. This change reflects that.

[1] https://github.com/gem5/gem5/pull/725

Change-Id: I7e72f5ea20d4aacf3115a485ba7cd664d33d037e
Signed-off-by: Hoa Nguyen <hn@hnpl.org>

powerjg reviewed (commented):
Is this going to break `semi` or the instruction-based ops?

hnpl commented:
I think it won't break instruction-based m5 ops as x86 does the same thing. I'm not sure about semi, though it doesn't look like semi uses m5_mem.

https://github.com/gem5/gem5/blob/9207458fd771e402cc98f67806dc83f5a0f79a0c/util/m5/src/abi/x86/SConsopts#L31

hnpl committed:
util-m5: Add a warning when m5op_addr is 0x0

This address, 0x0, is most likely a wrong address to call m5 ops.
The warning will catch the problem where m5op_addr is not initialized
properly.

Change-Id: I442b4806191ae6f5c137bc947f2a269684c599dd
Signed-off-by: Hoa Nguyen <hn@hnpl.org>
